tests/utils.go

```diff
@@ -1146,12 +1146,6 @@ func WaitForConfigToBePropagatedToComponent(podLabel string, resourceVersion str
 	}, duration, 1*time.Second).ShouldNot(HaveOccurred())
 }
 
-func RetryWithMetadataIfModified(objectMeta metav1.ObjectMeta, do func(objectMeta metav1.ObjectMeta) error) (err error) {
-	return RetryIfModified(func() error {
-		return do(objectMeta)
-	})
-}
-
 func RetryIfModified(do func() error) (err error) {
```

Comment:
What would you suggest to do with `RetryIfModified`? It's used by other 2 tests, let's create a new `retries` package for it?

Reply:
The fact that a test needs retries is a bad sign on its own right - tests should be deterministic. I would hide this in specific tests. Can you replace this with `Eventually` ?

Reply:
After we decided to use `Patch`, can we resolve this discussion?

Reply:
I'll resolve it, let me know if it should be reopened.